### DISCUSSION TITLE
KIF81HDD0 - Wrong doors state

### DISCUSSION THREAD

**Discussion by michal-ambroziak** [ekutner/home-connect-hass#158] [General]
I have Bosch refrigerator model KIF81HDD0 which shows that the doors are open when they are closed and vice versa.
For me, it looks like a problem with API, I reported this to Home Connect but never got any reply.
Can I temporarily fix this on the integration level?

binary_sensor.bosch_kif81hdd0_MAC_ADDRESS_bsh_common_status_doorstate

**Comment by ekutner** [comment]
You can use a template to reverse the state

**Comment by michal-ambroziak** [comment]
Is there a way without creating a new sensor?
Template sensor works, but it cannot be linked to Device Page https://community.home-assistant.io/t/add-template-sensors-to-device-page/193413 and due to that won't be processed properly by Dwains Dashboard.

I tried invert_logic: true under customize.yaml  but it doesn't work

**Reply by ekutner** [reply]
No other way that I'm aware of

**Comment by michal-ambroziak** [comment]
Would it be possible to implement inverted logic option for this integration if I raise a feature request?

**Reply by ekutner** [reply]
You can try installing the latest beta version and then add this configuration under home_connect_alt:
```
entity_settings:
  BSH.Common.Status.DoorState: 
    on_state: BSH.Common.EnumType.DoorState.Closed
```
I haven't tested it but it should work

**Reply by michal-ambroziak** [reply]
How can I do this only for one device?
In general, your solution works inverts state for all devices, so in the result, it fixes the problem for the refrigerator but creates an issue for other devices like the oven in my case.

**Reply by ekutner** [reply]
As you noticed this will apply to all your appliances. If you want it just for one you will have to use a template expression to do it. The power of the integration comes from being completely generic, so I won't add logic to for special cases.
